tests/system/test_observability.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2024 Google LLC All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import unittest
+import mock
+
+from . import _helpers
+from google.cloud.spanner_v1 import Client, DirectedReadOptions
+
+HAS_OTEL_INSTALLED = False
+
+try:
+    from opentelemetry.sdk.trace.export import SimpleSpanProcessor
+    from opentelemetry.sdk.trace.export.in_memory_span_exporter import (
+        InMemorySpanExporter,
+    )
+    from opentelemetry.sdk.trace import TracerProvider
+    from opentelemetry.sdk.trace.sampling import ALWAYS_ON
+    from opentelemetry import trace
+
+    HAS_OTEL_INSTALLED = True
+except ImportError:
+    pass
+
+
+@pytest.mark.skipif(not HAS_OTEL_INSTALLED, reason="OpenTelemetry needed.")
+@pytest.mark.skipif(not _helpers.USE_EMULATOR, reason="Emulator needed.")
+class TestObservability(unittest.TestCase):
```

Comment:
Don't make this a unit test. This is not qualified as unit test since we are actually using a server (emulator or backend) to execute the tests.

Please refer to these system tests that already use in-memory exporter to validate trace behavior.
I would recommend referring to these tests for verifiying spans in the other PR where we are refactoring spans .
https://github.com/googleapis/python-spanner/blob/b2c69d433a129a6313b83e9978a2b7bae73814cf/tests/system/test_session_api.py#L438

Reply:
So:
1. The tests being used to validate spans don't have a custom injection which makes them different than here
2. Very interesting that defining a class means they are "unittests", I thought this was more of just a way of organizing them?
3. I need to test 2 different scenarios for enable_extended_tracing which I've done with the new simple function call

Kindly help me take a look @harshachinta